services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/controller/SegmentRangeController.java

```diff
@@ -0,0 +1,174 @@
+package com.futurewei.alcor.vpcmanager.controller;
+
+import com.futurewei.alcor.common.entity.ResponseId;
+import com.futurewei.alcor.common.exception.ParameterNullOrEmptyException;
+import com.futurewei.alcor.common.exception.ResourceNotFoundException;
+import com.futurewei.alcor.common.exception.ResourceNullException;
+import com.futurewei.alcor.common.exception.ResourcePersistenceException;
+import com.futurewei.alcor.vpcmanager.service.SegmentRangeDatabaseService;
+import com.futurewei.alcor.vpcmanager.utils.RestPreconditionsUtil;
+import com.futurewei.alcor.web.entity.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.springframework.web.bind.annotation.RequestMethod.*;
+
+@RestController
+public class SegmentRangeController {
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Autowired
+    private SegmentRangeDatabaseService segmentRangeDatabaseService;
+
+    @RequestMapping(
+            method = GET,
+            value = {"/project/{projectid}/network_segment_ranges/{network_segment_range_id}", "/v4/{projectid}/network_segment_ranges/{network_segment_range_id}"})
+    public NetworkSegmentRangeWebResponseJson getSegmentRangeBySegmentRangeId(@PathVariable String projectid, @PathVariable String network_segment_range_id) throws Exception {
+
+        NetworkSegmentRangeWebResponseObject segmentRangeWebResponseObject = null;
+
+        try {
+            RestPreconditionsUtil.verifyParameterNotNullorEmpty(projectid);
+            RestPreconditionsUtil.verifyParameterNotNullorEmpty(network_segment_range_id);
+            RestPreconditionsUtil.verifyResourceFound(projectid);
+
+            segmentRangeWebResponseObject = this.segmentRangeDatabaseService.getBySegmentRangeId(network_segment_range_id);
+        } catch (ParameterNullOrEmptyException e) {
+            //TODO: REST error code
```

Comment:
I think it is time for us to use standard REST error code that is compatible to OpenStack.

Reply:
Okay ~

Reply:
@kevin-zhonghao Can you take a look at this commit: https://github.com/futurewei-cloud/alcor/pull/180/files#diff-17bb45b637cbf4c960554c2ae9fe82bc?

Specifically, all the exception classes has an associated error code so when we throw that exception, web layer will throw the associated error code. That is a good pattern to learn.

For example, 
services/port_manager/src/main/java/com/futurewei/alcor/portmanager/exception/ReleaseIpAddrBulkException.java

Reply:
Thanks for making the change!